utils/use_require.js

```diff
@@ -38,33 +38,58 @@ const no_transform_files = new Set([
 
 no_copy_files.forEach((file) => no_transform_files.add(file));
 
+function readFile(path) {
+    return new Promise((resolve, reject) => {
```

Comment:
https://nodejs.org/dist/latest-v9.x/docs/api/util.html#util_util_promisify_original

Reply:
basically everywhere with manual promises can be cleaned up with this

Reply:
plus, we should pre-declare promisified versions at the top for all promisified functions for readability inline

Reply:
Nice. I'll try to convert everything to that.

Reply:
Crap. No go. That requires a fairly new version of Node.js, and at least Fedora does not ship that. So I think it puts a too high requirement at this time.

I'll try to do something similar within the boundaries of  Node.js 6 though.